src/indexed_view.hpp

```diff
@@ -1,27 +1,26 @@
 #pragma once
 
-#include <vector/include/Vector.hpp>
+#include <memory/memory.hpp>
```

Comment:
It is now templated on `backend`. Because it is only used in mechanism implementations with the `memory::` library, I don't feel the need to make it more generic.

So I move it into the memory library.